### DISCUSSION TITLE
UTM is slower than multipass on Apple M1

### DISCUSSION THREAD

**Discussion by samchang8826** [utmapp/UTM#4779] [UTM Help (macOS)]
**Describe the issue**  
UTM is slower than multipass on Apple M1

**Configuration**  
* UTM Version: Version 4.0.9 (68)
* macOS Version: 13
* Mac Chip (Intel, M1, ...): Apple M1 Max

**Debug log**  

`sysbench --test=cpu run`

multipass:
![Screenshot 2022-12-07 at 23 59 01](https://user-images.githubusercontent.com/4272112/206228102-8844c6f9-f341-452b-9d8f-2a1c1bf87c71.png)

UTM
![Screenshot 2022-12-08 at 00 00 59](https://user-images.githubusercontent.com/4272112/206228516-537a7147-b94e-4a77-9738-ee25df79b32d.png)

**Comment by osy** [comment]
I’m not sure what the issue is. Benchmarks will always be slower in a VM than on native because of overheads.

**Reply by samchang8826** [reply]
multipass is using qemu too, not native

```
$ /Library/Application\ Support/com.canonical.multipass/bin/qemu-system-aarch64 -version
QEMU emulator version 6.2.0
Copyright (c) 2003-2021 Fabrice Bellard and the QEMU Project developers
```

**Reply by osy** [reply]
Is it possible to dump what arguments they’re using? You can then compare. I’m wondering if the number of cores specified is different.

**Reply by samchang8826** [reply]
FYI

```
$ps aux | grep multipass

/Library/Application Support/com.canonical.multipass/bin/qemu-system-aarch64 -machine virt,highmem=off -accel hvf -drive file=/Library/Application Support/com.canonical.multipass/bin/../Resources/qemu/edk2-aarch64-code.fd,if=pflash,format=raw,readonly=on -cpu cortex-a72 -nic vmnet-shared,model=virtio-net-pci,mac=52:54:00:0d:3d:51 -device virtio-scsi-pci,id=scsi0 -drive file=/var/root/Library/Application Support/multipassd/qemu/vault/instances/xxxx/ubuntu-20.04-server-cloudimg-arm64.img,if=none,format=qcow2,discard=unmap,id=hda -device scsi-hd,drive=hda,bus=scsi0.0 -smp 8 -m 8192M -qmp stdio -chardev null,id=char0 -serial chardev:char0 -nographic -cdrom /var/root/Library/Application Support/multipassd/qemu/vault/instances/xxxx/cloud-init-config.iso -loadvm suspend -machine virt-6.2
```

**Reply by osy** [reply]
Did you set your VM also to 8 cores and 8GB ram?

**Reply by samchang8826** [reply]
yes, i have also tried to set cpu as `cortex-a72`, the result does not change

**Reply by samchang8826** [reply]
I was wondering if you could reproduce this issue by installing multipass on your local machine ...

multipass is a tool provided by Ubuntu (Canonical Ltd.)

**Reply by osy** [reply]
@samchang8826 I just ran it with my M1 MacBook Air (set to 4G RAM, 4 cores) and got the opposite result. Using `sysbench --test=cpu run` I got 86203 number of events on multipass and 109167 on UTM.

Also, the benchmark seems to be single threaded so CPU count shouldn't matter. Not sure if that's intentional. I think what may be happening is that macOS is deciding to schedule the benchmark on an E-core randomly.
